### ISSUE TITLE
Calls with more than 2 parties with video fail to reconnect one or more

### ISSUE THREAD

**Issue by tdm4** [nextcloud/spreed#2337]
## Steps to reproduce
1. Start a call with another party (both have video and audio enabled)
2. Wait for third person to join
3. Have third person enable video

### Expected behaviour
All 3 parties should be able to see and hear each other

### Actual behaviour
If Party A and Party B are having a video+audio conversation, and Party C joins. If they unmute their microphone, it's fine, all 3 parties can hear each other. If Party C then tries to enable their video, then either Party A cannot ever see Party C (Party A sees endless spinning circle from Party C), and often Party B can see and hear both Party A and C, but A and C cannot see each other.

It's 3-way video that is failing. A TURN server is configured and NextCloud can talk to it, also all the high UDP relay ports are open, so it should all be working.

## Browser
Party A = Firefox
Party B = NextCloud Talk Android App
Party C = Chrome

Have also tried (on Party A) latest Chrome as well  77.0.3865.120 and still get same behavior

**Microphone available:** yes

**Camera available:** yes

**Operating system:** 
Party A = Void Linux
Party B = Android
Party C = Ubuntu Linux

**Browser name:** 
Party A = FIrefox or Chromium
Party B = NextCloud Talk Android (not using browser)
Party C = Chromium

**Browser version:** 
Chromium 77
Firefox 69

### Browser log
Here is Console data from Party A's browser (Chromium 77)
<details>

```
merged.js?v=2a6c862b-0:16524 localMediaStarted
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
tooltip.js:28 Deprecated call $.tooltip('destroy') has been deprecated and should be removed
e.tooltip @ tooltip.js:28
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:15981 Connecting to peer...
merged.js?v=2a6c862b-0:16603 VIDEO ADDED Peer
merged.js?v=2a6c862b-0:16603 VIDEO ADDED Peer
merged.js?v=2a6c862b-0:15987 Connection established.
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 simplewebrtc datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:15981 Connecting to peer...
merged.js?v=2a6c862b-0:15987 Connection established.
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 simplewebrtc datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:15699 XXX Remove peer 3K35YSxwWlvzIhMQ9RAFqbMjxy/2A/48xn9x2NlviGONS+CnPE/yifnmqCwrQTXi5sgslnl5BLHLU8iCDY+tPHQMneIvlP3ADpMcHBattWFlVnDlwWT9Mnzde2yVhzO+EIlCpiq4mOWdI6XQUgTIRYUYTvhlR/9MZvDI+mByog605GssvDTtyhoE8cju9SdXvonnUT5RPYWLCibMU2M4JqEYfR0q1kIarRNc7b7YmMAjAilQ1rXwBnu4s0mxskw
merged.js?v=2a6c862b-0:16074 Connection closed.
merged.js?v=2a6c862b-0:15699 XXX Remove peer fxtbJ6Mes0n+AvmjQMxSlOXuAz3NsWUAq1v4Z/3AoGrm2ZX1LNb+IdGWhBqj2aZvYcT6gMfHOArs1xWs/i7FDwVc2wH2NtCXIXlJF8HVXd7IgpmRzG/EhkH1Q1fi+U5txS/bcE/h8hbI03WW4Dp0jQwJBXHafTno7pZzR0FHZ62eXcNNzuzNomDIt7DtD5ruSIOwOdiYSk0sFbsIMB8L5FjykVylYR2IU6bXWT/6v3dB/uUXRSfcJFymaK5No4J
merged.js?v=2a6c862b-0:16074 Connection closed.
merged.js?v=2a6c862b-0:17371 Forced reconnects with a new session are not supported in the internal signaling; same session as before will be used
jquery.js:8561 [Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
send @ jquery.js:8561
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:15981 Connecting to peer...
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
merged.js?v=2a6c862b-0:16603 VIDEO ADDED Peer
merged.js?v=2a6c862b-0:16074 Connection closed.
merged.js?v=2a6c862b-0:16074 Connection closed.
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
tooltip.js:28 Deprecated call $.tooltip('destroy') has been deprecated and should be removed
e.tooltip @ tooltip.js:28
merged.js?v=2a6c862b-0:15699 XXX Remove peer fxtbJ6Mes0n+AvmjQMxSlOXuAz3NsWUAq1v4Z/3AoGrm2ZX1LNb+IdGWhBqj2aZvYcT6gMfHOArs1xWs/i7FDwVc2wH2NtCXIXlJF8HVXd7IgpmRzG/EhkH1Q1fi+U5txS/bcE/h8hbI03WW4Dp0jQwJBXHafTno7pZzR0FHZ62eXcNNzuzNomDIt7DtD5ruSIOwOdiYSk0sFbsIMB8L5FjykVylYR2IU6bXWT/6v3dB/uUXRSfcJFymaK5No4J
merged.js?v=2a6c862b-0:15699 XXX Remove peer 3K35YSxwWlvzIhMQ9RAFqbMjxy/2A/48xn9x2NlviGONS+CnPE/yifnmqCwrQTXi5sgslnl5BLHLU8iCDY+tPHQMneIvlP3ADpMcHBattWFlVnDlwWT9Mnzde2yVhzO+EIlCpiq4mOWdI6XQUgTIRYUYTvhlR/9MZvDI+mByog605GssvDTtyhoE8cju9SdXvonnUT5RPYWLCibMU2M4JqEYfR0q1kIarRNc7b7YmMAjAilQ1rXwBnu4s0mxskw
merged.js?v=2a6c862b-0:16524 localMediaStarted
tooltip.js:28 Deprecated call $.tooltip('destroy') has been deprecated and should be removed
e.tooltip @ tooltip.js:28
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:15981 Connecting to peer...
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
merged.js?v=2a6c862b-0:16603 VIDEO ADDED Peer
merged.js?v=2a6c862b-0:15981 Connecting to peer...
merged.js?v=2a6c862b-0:16603 VIDEO ADDED Peer
merged.js?v=2a6c862b-0:15987 Connection established.
merged.js?v=2a6c862b-0:15987 Connection established.
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 simplewebrtc datachannel is open
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 simplewebrtc datachannel is open
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:15699 XXX Remove peer fxtbJ6Mes0n+AvmjQMxSlOXuAz3NsWUAq1v4Z/3AoGrm2ZX1LNb+IdGWhBqj2aZvYcT6gMfHOArs1xWs/i7FDwVc2wH2NtCXIXlJF8HVXd7IgpmRzG/EhkH1Q1fi+U5txS/bcE/h8hbI03WW4Dp0jQwJBXHafTno7pZzR0FHZ62eXcNNzuzNomDIt7DtD5ruSIOwOdiYSk0sFbsIMB8L5FjykVylYR2IU6bXWT/6v3dB/uUXRSfcJFymaK5No4J
merged.js?v=2a6c862b-0:16074 Connection closed.
merged.js?v=2a6c862b-0:15699 XXX Remove peer 3K35YSxwWlvzIhMQ9RAFqbMjxy/2A/48xn9x2NlviGONS+CnPE/yifnmqCwrQTXi5sgslnl5BLHLU8iCDY+tPHQMneIvlP3ADpMcHBattWFlVnDlwWT9Mnzde2yVhzO+EIlCpiq4mOWdI6XQUgTIRYUYTvhlR/9MZvDI+mByog605GssvDTtyhoE8cju9SdXvonnUT5RPYWLCibMU2M4JqEYfR0q1kIarRNc7b7YmMAjAilQ1rXwBnu4s0mxskw
merged.js?v=2a6c862b-0:16074 Connection closed.
merged.js?v=2a6c862b-0:17371 Forced reconnects with a new session are not supported in the internal signaling; same session as before will be used
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:17432 OFFER Object
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
merged.js?v=2a6c862b-0:15981 Connecting to peer...
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
merged.js?v=2a6c862b-0:16603 VIDEO ADDED Peer
merged.js?v=2a6c862b-0:15981 Connecting to peer...
merged.js?v=2a6c862b-0:16024 Disconnected.
merged.js?v=2a6c862b-0:16044 ICE restart.
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:15981 Connecting to peer...
merged.js?v=2a6c862b-0:15987 Connection established.
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 simplewebrtc datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:16074 Connection closed.
merged.js?v=2a6c862b-0:16074 Connection closed.
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
tooltip.js:28 Deprecated call $.tooltip('destroy') has been deprecated and should be removed
e.tooltip @ tooltip.js:28
merged.js?v=2a6c862b-0:15699 XXX Remove peer 3K35YSxwWlvzIhMQ9RAFqbMjxy/2A/48xn9x2NlviGONS+CnPE/yifnmqCwrQTXi5sgslnl5BLHLU8iCDY+tPHQMneIvlP3ADpMcHBattWFlVnDlwWT9Mnzde2yVhzO+EIlCpiq4mOWdI6XQUgTIRYUYTvhlR/9MZvDI+mByog605GssvDTtyhoE8cju9SdXvonnUT5RPYWLCibMU2M4JqEYfR0q1kIarRNc7b7YmMAjAilQ1rXwBnu4s0mxskw
merged.js?v=2a6c862b-0:15699 XXX Remove peer fxtbJ6Mes0n+AvmjQMxSlOXuAz3NsWUAq1v4Z/3AoGrm2ZX1LNb+IdGWhBqj2aZvYcT6gMfHOArs1xWs/i7FDwVc2wH2NtCXIXlJF8HVXd7IgpmRzG/EhkH1Q1fi+U5txS/bcE/h8hbI03WW4Dp0jQwJBXHafTno7pZzR0FHZ62eXcNNzuzNomDIt7DtD5ruSIOwOdiYSk0sFbsIMB8L5FjykVylYR2IU6bXWT/6v3dB/uUXRSfcJFymaK5No4J
merged.js?v=2a6c862b-0:16524 localMediaStarted
tooltip.js:28 Deprecated call $.tooltip('destroy') has been deprecated and should be removed
e.tooltip @ tooltip.js:28
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
merged.js?v=2a6c862b-0:15677 Starting call with Object
merged.js?v=2a6c862b-0:16350 PEER CREATED Peer
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:15981 Connecting to peer...
merged.js?v=2a6c862b-0:15981 Connecting to peer...
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
q46iywqk#:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
merged.js?v=2a6c862b-0:16603 VIDEO ADDED Peer
merged.js?v=2a6c862b-0:16603 VIDEO ADDED Peer
merged.js?v=2a6c862b-0:15987 Connection established.
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 simplewebrtc datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:16024 Disconnected.
merged.js?v=2a6c862b-0:16044 ICE restart.
merged.js?v=2a6c862b-0:17432 OFFER Object
merged.js?v=2a6c862b-0:15981 Connecting to peer...
merged.js?v=2a6c862b-0:15987 Connection established.
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 simplewebrtc datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:16569 status datachannel is open
merged.js?v=2a6c862b-0:16569 hark datachannel is open
merged.js?v=2a6c862b-0:16074 Connection closed.
merged.js?v=2a6c862b-0:16074 Connection closed.
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
merged.js?v=2a6c862b-0:15699 XXX Remove peer fxtbJ6Mes0n+AvmjQMxSlOXuAz3NsWUAq1v4Z/3AoGrm2ZX1LNb+IdGWhBqj2aZvYcT6gMfHOArs1xWs/i7FDwVc2wH2NtCXIXlJF8HVXd7IgpmRzG/EhkH1Q1fi+U5txS/bcE/h8hbI03WW4Dp0jQwJBXHafTno7pZzR0FHZ62eXcNNzuzNomDIt7DtD5ruSIOwOdiYSk0sFbsIMB8L5FjykVylYR2IU6bXWT/6v3dB/uUXRSfcJFymaK5No4J
merged.js?v=2a6c862b-0:15699 XXX Remove peer 3K35YSxwWlvzIhMQ9RAFqbMjxy/2A/48xn9x2NlviGONS+CnPE/yifnmqCwrQTXi5sgslnl5BLHLU8iCDY+tPHQMneIvlP3ADpMcHBattWFlVnDlwWT9Mnzde2yVhzO+EIlCpiq4mOWdI6XQUgTIRYUYTvhlR/9MZvDI+mByog605GssvDTtyhoE8cju9SdXvonnUT5RPYWLCibMU2M4JqEYfR0q1kIarRNc7b7YmMAjAilQ1rXwBnu4s0mxskw
tooltip.js:28 Deprecated call $.tooltip('destroy') has been deprecated and should be removed
e.tooltip @ tooltip.js:28
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
globals.js:30 The escapeHTML library is deprecated! It will be removed in nextcloud 19.
i @ globals.js:30
```
</details>

## Talk app

**Talk app version:**
6.0.4

**Custom TURN server configured:** yes

**Custom STUN server configured:** yes

## Server configuration

**Operating system**:  OpenBSD 6.6

**Web server:**   OpenBSD httpd

**Database:** PostgreSQL 11.4

**PHP version:**   7.3.10

**Nextcloud Version:**  16.0.4

**List of activated apps:**

<details>

```
Enabled:
  - accessibility: 1.2.0
  - activity: 2.9.1
  - cloud_federation_api: 0.2.0
  - comments: 1.6.0
  - contacts: 3.1.6
  - dav: 1.9.2
  - federatedfilesharing: 1.6.0
  - federation: 1.6.0
  - files: 1.11.0
  - files_pdfviewer: 1.5.0
  - files_rightclick: 0.15.1
  - files_sharing: 1.8.0
  - files_texteditor: 2.8.0
  - files_trashbin: 1.6.0
  - files_versions: 1.9.0
  - files_videoplayer: 1.5.0
  - firstrunwizard: 2.5.0
  - gallery: 18.3.0
  - gpxmotion: 0.0.10
  - gpxpod: 4.0.5
  - logreader: 2.1.0
  - lookup_server_connector: 1.4.0
  - nextcloud_announcements: 1.5.0
  - notifications: 2.4.1
  - oauth2: 1.4.2
  - password_policy: 1.6.0
  - passwords: 2019.10.0
  - privacy: 1.0.0
  - provisioning_api: 1.6.0
  - recommendations: 0.4.0
  - serverinfo: 1.6.0
  - sharebymail: 1.6.0
  - spreed: 6.0.4
  - support: 1.0.0
  - survey_client: 1.4.0
  - systemtags: 1.6.0
  - theming: 1.7.0
  - twofactor_backupcodes: 1.5.0
  - twofactor_totp: 3.0.1
  - updatenotification: 1.6.0
  - viewer: 1.1.0
  - workflowengine: 1.6.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_ldap
```
</details>

**Nextcloud configuration:**

<details>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.pertho.net"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/xxxx",
        "overwritehost": "xxxx",
        "overwriteprotocol": "https",
        "dbtype": "pgsql",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "version": "16.0.4.1",
        "logtimezone": "Europe\/London",
        "installed": true,
        "default_language": "en",
        "defaultapp": "files",
        "knowledgebaseenabled": true,
        "allow_user_to_change_display_name": true,
        "remember_login_cookie_lifetime": 1296000,
        "session_lifetime": 86400,
        "session_keepalive": true,
        "token_auth_enforced": false,
        "auth.bruteforce.protection.enabled": true,
        "mail_smtpmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "has_internet_connection": true,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": 0
        },
        "loglevel": 0,
        "theme": "",
        "maintenance": false,
        "mail_smtpsecure": "tls",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "integrity.check.disabled": true,
        "has_rebuilt_cache": true,
        "app_install_overwrite": [
            "calendar"
        ],
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***"
    }
}
```
</details>

### Server log (data/nextcloud.log)
<details>

```
Not applicable -- only cron logs in there
```
</details>

**Comment by tdm4** [comment]
OK I think I am onto something.. note the configuration:

Operating system:
Party A = Void Linux
Party B = Android
Party C = Ubuntu Linux

Browser name:
Party A = FIrefox or Chromium
Party B = NextCloud Talk Android (not using browser)
Party C = Chromium

If Party A starts the calll, and Party B joins.. call is fine. When Party C tries to join, they can only enable audio. If they enable video they spend forever connecting.

BUT!

If Party A starts the call and Party C joins first, then Party B can join afterwards and the 3-way call works fine.

One thing is, Party B (On the Nextcloud Talk android app), their video/image never switches to the "main" screen in the browser when they speak. This might be some missing signalling to the NextCloud Server that the browsers somehow do. (But I think this is a separate issue.)

Hope this helps!

**Comment by tdm4** [comment]
Since I filed this bug, I am now on NextCloud 17.0.0 and NextTalk 7.0.1, but the same behavior still happens.

**Comment by nickvergessen** [comment]
Is there any useful output in your javascript console (Press F12)

**Comment by tdm4** [comment]
@nickvergessen I'll try to get some but I didn't see anything that useful. I can see it try to connect but then it says disconnect ICE.. but no error. I'll try and get thefull console output next time I try which should be this evening.

You should be able to reproduce by using 3 parties: 2 people with firefox or chrome, and 1 using the android nextcloud talk app. Have one of the parties with the browser join first, then have the talk app person join, then try to get the 3rd person with the browser to join after that. It will fail to work.

If the 2 people with the browsers join the call first, then there are no problems when the android nextcloud talk app person joins.

**Comment by tdm4** [comment]
OK in 17.0.1 and NextCloud Talk 7.0.1, it seems a lot smoother. Sometimes if one person drops out of the call and rejoins, their video doesn't always reconnect. This is some of the messages:
```
merged-guest.js?v=ac6f87c9-0:16488 Starting call with {userId: "tom", roomId: 19, lastPing: 1574595059, sessionId: "YKuZM7QTEd1CqwM6Sw5DYQRRfAFoFXKcg+IQYbOfnAL
tmxCnML…CW7jfVNqFRA8k54ieZ4S/EfmG2q0H1XDQzp535SvJ49ljlahL", inCall: 7}
merged-guest.js?v=ac6f87c9-0:17161 PEER CREATED Peer {id: "YKuZM7QTEd1CqwM6Sw5DYQRRfAFoFXKcg+IQYbOfnALtmxCnML…CW7jfVNqFRA8k54ieZ4S/EfmG2q0H1XDQzp535SvJ49ljlahL
", parent: WebRTC, type: "video", oneway: false, sharemyscreen: false, …}
merged-guest.js?v=ac6f87c9-0:18243 OFFER {to: "YKuZM7QTEd1CqwM6Sw5DYQRRfAFoFXKcg+IQYbOfnALtmxCnML…CW7jfVNqFRA8k54ieZ4S/EfmG2q0H1XDQzp535SvJ49ljlahL", sid: "157
4595059966", broadcaster: undefined, roomType: "video", type: "offer", …}
merged-guest.js?v=ac6f87c9-0:16792 Connecting to peer...
merged-guest.js?v=ac6f87c9-0:17161 PEER CREATED Peer {id: "u9Qs26V5gJic1w9eVswIDp8DmJhTGg4VNu9ar8KvlHO44vGgvn…E5f9TE9Z9PUPkyC3j/AieEw4QZvDxSUUzYNO43TVX3jrCMovs
", parent: WebRTC, type: "video", oneway: false, sharemyscreen: false, …}
merged-guest.js?v=ac6f87c9-0:17414 VIDEO ADDED Peer {id: "u9Qs26V5gJic1w9eVswIDp8DmJhTGg4VNu9ar8KvlHO44vGgvn…E5f9TE9Z9PUPkyC3j/AieEw4QZvDxSUUzYNO43TVX3jrCMovs"
, parent: WebRTC, type: "video", oneway: false, sharemyscreen: false, …}
merged-guest.js?v=ac6f87c9-0:18241 ANSWER {to: "u9Qs26V5gJic1w9eVswIDp8DmJhTGg4VNu9ar8KvlHO44vGgvn…E5f9TE9Z9PUPkyC3j/AieEw4QZvDxSUUzYNO43TVX3jrCMovs", sid: "15
74595155758", broadcaster: undefined, roomType: "video", type: "answer", …}
merged-guest.js?v=ac6f87c9-0:16792 Connecting to peer...
merged-guest.js?v=ac6f87c9-0:16798 Connection established.
pysjpkvt:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
merged-guest.js?v=ac6f87c9-0:17414 VIDEO ADDED Peer {id: "YKuZM7QTEd1CqwM6Sw5DYQRRfAFoFXKcg+IQYbOfnALtmxCnML…CW7jfVNqFRA8k54ieZ4S/EfmG2q0H1XDQzp535SvJ49ljlahL", parent: WebRTC, type: "video", oneway: false, sharemyscreen: false, …}
merged-guest.js?v=ac6f87c9-0:16798 Connection established.
merged-guest.js?v=ac6f87c9-0:17380 status datachannel is open
merged-guest.js?v=ac6f87c9-0:17380 simplewebrtc datachannel is open
merged-guest.js?v=ac6f87c9-0:17380 status datachannel is open
merged-guest.js?v=ac6f87c9-0:17380 status datachannel is open
merged-guest.js?v=ac6f87c9-0:17380 simplewebrtc datachannel is open
```

It seems to error adding ICE Candidate, but then it adds the video stream but I think it's only one way. Unless the party who can't see the video leaves the call and rejoins.

It seems to recover faster setting up the 3-way video calls though than it used to.

I'm not sure how else I can get an accurate log to help. Any suggestions?

**Comment by tdm4** [comment]
Had something interesting happen tonight.

I joined the call (no one else was in it) and it started.. Call me "Party A"
Party B joins.. they're using chromium (on Linux). Works when they only have audio enabled.
I had both video and audio enabled.

When they enabled their video, I just get a circle spinning for their name.
Then, I left them alone in the call. (I clicked 'Leave Call') and force-refreshed the web page with the call, then rejoined.

THEN, I could see and hear Party B. Party C joined a minute or two later using the Android app and all worked fine.

Only issue was, when Party C spoke, the focus main screen never shifted to them, it was always on Party B.

Other than that, the call went fine.

So steps that worked:
1) Party A starts call
2) Party B joins call, enables audio and video
3) Party A and B can't see or hear each other (endless spinning circle)
4) Party A leaves call, forcibly refreshes page and rejoins call
5) Enabled video and audio and all worked well from then on.

I have no idea why it just does work the first time?

**Comment by tdm4** [comment]
Another thing I noticed.. when all 3 parties were on and connected there were 16 redis keys in Coturn TURN server's 'stats' database.

They all looked like this:
```
 1) "turn/realm/cloud.pertho.net/user/1575918691:uaQB7AtJvL1TW1bS/allocation/000000000000000038/status"
 2) "turn/realm/cloud.pertho.net/user/1575918691:uaQB7AtJvL1TW1bS/allocation/001000000000000017/status"
 3) "turn/realm/cloud.pertho.net/user/1575918691:uaQB7AtJvL1TW1bS/allocation/001000000000000016/status"
 4) "turn/realm/cloud.pertho.net/user/1575918212:FMcfegcLVtyaRuEL/allocation/001000000000000022/status"
 5) "turn/realm/cloud.pertho.net/user/1575918390:mpqS6jPUyfno8j9H/allocation/001000000000000015/status"
 6) "turn/realm/cloud.pertho.net/user/1575918390:mpqS6jPUyfno8j9H/allocation/000000000000000037/status"
 7) "turn/realm/cloud.pertho.net/user/1575918390:mpqS6jPUyfno8j9H/allocation/000000000000000036/status"
 8) "turn/realm/cloud.pertho.net/user/1575918691:uaQB7AtJvL1TW1bS/allocation/001000000000000025/status"
 9) "turn/realm/cloud.pertho.net/user/1575918691:uaQB7AtJvL1TW1bS/allocation/001000000000000024/status"
10) "turn/realm/cloud.pertho.net/user/1575918691:uaQB7AtJvL1TW1bS/allocation/001000000000000023/status"
11) "turn/realm/cloud.pertho.net/user/1575918390:mpqS6jPUyfno8j9H/allocation/000000000000000021/status"
12) "turn/realm/cloud.pertho.net/user/1575918212:FMcfegcLVtyaRuEL/allocation/000000000000000018/status"
13) "turn/realm/cloud.pertho.net/user/1575918691:uaQB7AtJvL1TW1bS/allocation/001000000000000019/status"
14) "turn/realm/cloud.pertho.net/user/1575918212:FMcfegcLVtyaRuEL/allocation/001000000000000011/status"
15) "turn/realm/cloud.pertho.net/user/1575918212:FMcfegcLVtyaRuEL/allocation/001000000000000018/status"
16) "turn/realm/cloud.pertho.net/user/1575918691:uaQB7AtJvL1TW1bS/allocation/000000000000000028/status"
```

Kind of strange why it'd generate all those.

**Comment by nickvergessen** [comment]
Maybe ask the coturn guys?

**Comment by tdm4** [comment]
Hi @nickvergessen - It doesn't an issue now. I just upgraded to Nextcloud 18.0.0 so I'm going to try another 3-way talk tomorrow, but the past few weeks I've used it, it's not had an issue. Sometimes it takes a little while to set up when the 3rd person joins, but I think there's some network routing packet drops between one of the parties and the nextcloud talk server, so that could be why it takes a little longer to set up.
